tests/test_tools.py

```diff
@@ -0,0 +1,30 @@
+"""Unit tests for oauth2client.tools."""
+
+import unittest
+from oauth2client import tools
+from six.moves.urllib import request
+import threading
+
+class TestClientRedirectServer(unittest.TestCase):
+    """Test the ClientRedirectServer and ClientRedirectHandler classes."""
+
+    def test_ClientRedirectServer(self):
+        # create a ClientRedirectServer and run it in a thread to listen
+        # for a mock GET request with the access token
+        # the server should return a 200 message and store the token
+        httpd = tools.ClientRedirectServer(('localhost', 0), tools.ClientRedirectHandler)
+        code = 'foo'
+        url = 'http://localhost:%i?code=%s' % (httpd.server_address[1], code)
+        t = threading.Thread(target = httpd.handle_request)
+        t.setDaemon(True)
```

Comment:
Thinking of it a little more, it helps if the main thread crashes before finishing connection to the handler thread, in which case the test app will not wait forever for the thread to close.